### ISSUE TITLE
`JavaPropsGenerator#writeFieldName()` should not escape property keys

### ISSUE THREAD

**Issue by dmandalidis** [FasterXML/jackson-dataformats-text#91]
I think that ```JPropEscapes.appendKey(_basePath, name);``` in ```JavaPropsGenerator#writeFieldName``` should be moved *only* to ```WriterBackedGenerator```. Escaping keys is a serialization matter when a Properties object is written to something other than memory. So, if you do:

```
public class TestObject {
	Map<String, String> values = new HashMap<>();
	public Map<String, String> getValues() {
		return values;
	}
	public void setValues(Map<String, String> values) {
		this.values = values;
	}
}
[...]
@Test
	public void test() throws Exception {
		TestObject expected = new TestObject();
		expected.values.put("foo:bar", "1");
		Properties properties = mapper.writeValueAsProperties(expected);
		TestObject actual = mapper.readPropertiesAs(properties, TestObject.class);
		assertThat(actual.values).isEqualTo(expected.values);
	}
```

This fails because the properties object already contains the key escaped as ```foo\:bar```, but readProperties expects that it doesn't. The properties itself, will add another backslash if it's about to be written to an actual file.

**Comment by cowtowncoder** [comment]
I am not sorry I follow your reasoning here distinction.

**Comment by dmandalidis** [comment]
Well, I think that JavaPropsMapper#writeValueAsProperties shouldn't not escape special characters from property keys for the same reason Properties#put method doesn't. (Un)escaping such characters is needed only when the underlying Properties object is (de)serialized, and that's why it originally happens only in Properties#load and Properties#store.

As far as this issue is concerned, you have a correct distinction between these two; PropertiesBackedGenerator when writing to a properties object and WriterBackedGenerator when writing to an output stream. You 're correctly using PropertiesBackedGenerator for JavaPropsMapper#writeValueAsProperties and WriterBackedGenerator for ObjectMapper#writeValue. However, their base method JavaPropsGenerator#writeFieldName doesn't make such a distinction, and unconditionally applies escaping no matter where the object in question is written.

So, I believe that:
- ```JPropEscapes.appendKey(_basePath, name);``` should be removed from ```JavaPropsGenerator#writeFieldName```
- ```JavaPropsGenerator#writeFieldName``` should be overridden by ```WriterBackedGenerator``` by calling the base implementation plus the ```JPropEscapes.appendKey(_basePath, name);``` statement.

I 'll send a PR to make it clearer

**Comment by cowtowncoder** [comment]
@dmandalidis Ok, apologies for slow follow up here: I think I see your point. Will try to have a proper look soon; adding this to my todo list.

**Comment by dmandalidis** [comment]
@cowtowncoder no worries

**Comment by cowtowncoder** [comment]
Yes, absolutely right, escaping should not be done for keys or values. Fixed for 2.9.9, later versions.
